examples/pytorch/question-answering/sparseml_utils.py

```diff
@@ -106,6 +138,22 @@ def compute_loss(self, model, inputs, return_outputs=False):
             loss = ((1 - self.distill_hardness) * label_loss) + (self.distill_hardness * teacher_loss)
         return (loss, outputs) if return_outputs else loss
 
+    def save_model(self, output_dir: Optional[str] = None):
+        """
+        Save model during or after training. The sparsification recipe will also be saved.
+        """
+        super().save_model(output_dir=output_dir)
+        self._save_recipe(output_dir=output_dir)
+
+    def _save_recipe(self, output_dir: Optional[str] = None):
+        if output_dir is None:
+            output_dir = self.args.output_dir
+        output_dir = output_dir if output_dir is not None else self.args.output_dir
+        os.makedirs(output_dir, exist_ok=True)
+        output_recipe_file = os.path.join(output_dir, RECIPE_NAME)
+        with open(output_recipe_file, "w") as fp:
+            json.dump({"recipe": str(self.manager) if self.manager is not None else None}, fp)
```

Comment:
is there a reason we are saving the entire string representation of the manager, instead of the YAML recipe by using `manager.save()`?

Reply:
Fix, now recipe is saved as yaml